src/jit/codegenxarch.cpp

```diff
@@ -6474,7 +6474,7 @@ void CodeGen::genJmpMethod(GenTreePtr jmp)
             }
         }
 
-#if FEATURE_VARARG
+#if FEATURE_VARARG && defined(_TARGET_AMD64_)
```

Comment:
The logic in 6380-6386 is ok for x86; it does no harm. In Compiler::lvaPromoteStructVar(), when we promote a struct, only for amd64/arm64 will a lvIsRegArg struct have its promoted fields marked as lvIsRegArg=true. But for x86, the struct itself is never marked lvIsRegArg even through it can be promoted (with a single field). So in this code we can load the promoted field varDsc, but we'll immediately "continue".